### PR TITLE
(HI-486) simplify acceptance setup

### PR DESCRIPTION
**Pull request by joshcooper**
This commit removes support for `packages` type of acceptance, bumps our beaker dependency, and uses beaker's updated `install_puppet_agent_dev_repo_on` method to install puppet-agent on all platforms.

### CODE REVIEW

File: acceptance/Gemfile [hunk 1]
```diff
@@ -10,7 +10,7 @@ def location_for(place, fake_version = nil)
   end
 end
 
-gem "beaker", *location_for(ENV['BEAKER_VERSION'] || "~> 2.27")
+gem "beaker", *location_for(ENV['BEAKER_VERSION'] || "~> 2.31")
```
Comment (joshcooper) [inline acceptance/Gemfile]: This PR can't be merged until there's a beaker release containing the fix for https://github.com/puppetlabs/beaker/pull/1044, and this constraint may need to be updated e.g. `2.32`.

**Comment by joshcooper** [comment]
@johnduarte @er0ck Passed on all platforms (except cisco) using beaker 2.23.0:

```
SHA=5c59c61a4d02ffb32135c9eb327136c7ab64bbca
SUITE_VERSION=1.3.2.216.g5c59c61

test_targets=(
        arista4-32a 
        centos4-64a centos4-32a
        cumulus25-64a
        debian6-64a debian6-32a
        debian7-64a debian7-32a
        debian8-64a debian8-32a
        fedora21-64a fedora21-32a
        fedora22-64a fedora22-32a
        osx109-64a
        osx1010-64a
        osx1011-64a
        redhat5-64a redhat5-32a
        redhat6-64a redhat6-32a
        redhat7-64a
        sles10-64a sles10-32a
        sles11-64a sles11-32a
        sles12-64a
        solaris10-64a
        solaris11-64a
        ubuntu1004-64a ubuntu1004-32a
        ubuntu1204-64a ubuntu1204-32a
        ubuntu1404-64a ubuntu1404-32a
        ubuntu1504-64a ubuntu1504-32a
        ubuntu1510-64a ubuntu1510-32a
        windows2008r2-64a windows2008r2-6432a
        windows2012r2-64a windows2012r2-6432a )
```

File: acceptance/lib/puppet/acceptance/install_utils.rb [hunk 2]
```diff
@@ -53,121 +53,6 @@ def install_packages_on(hosts, package_hash, options = {})
         end
         return true
       end
-
```
Comment (MikaelSmith) [inline acceptance/lib/puppet/acceptance/install_utils.rb]: Why do we still need install_packages_on?

**Comment by MikaelSmith** [comment]
:+1:

File: acceptance/setup/aio/pre-suite/010_Install.rb [hunk 3]
```diff
@@ -1,35 +1,18 @@
...
 
-step "Install repositories on target machines..." do
-  sha = ENV['SHA']
-  hosts.each do |host|
-    install_repos_on(host, 'puppet-agent', sha, 'repo-configs')
-  end
+step "Install puppet-agent..." do
+  opts = {
+    :puppet_collection    => 'PC1',
+    :puppet_agent_sha     => ENV['SHA'],
```
Comment (er0ck) [inline acceptance/setup/aio/pre-suite/010_Install.rb]: many jobs use PACKAGE_BUILD_VERSION.  should it be used here as well?

File: acceptance/lib/puppet/acceptance/install_utils.rb (hunk 2 above)
Comment (er0ck) [inline acceptance/lib/puppet/acceptance/install_utils.rb]: does the git method currently work?  i'd expect it does, because, hiera.

**Pull request merged**